checker/src/summaries.rs

```diff
@@ -598,10 +601,12 @@ impl<'a, 'tcx: 'a> PersistentSummaryCache<'a, 'tcx> {
     ) -> &Summary {
         let dependents = self.dependencies.entry(def_id).or_insert_with(Vec::new);
         if !dependents.contains(&dependent_def_id) {
+            assume!(dependents.len() < std::usize::MAX);
             dependents.push(dependent_def_id);
         }
         if self.typed_cache.contains_key(&function_id) {
-            self.typed_cache.get(&function_id).unwrap()
+            let result = self.typed_cache.get(&function_id);
+            result.expect("value disappeared from typed_cache")
```

Comment:
Instead of this, perhaps we can just do `if let Some(result) = self.typed_cache.get(&function_id) { ... } else { ... }`?

Reply:
The point of this is to suggest that this failure can only happen if the code is changed incorrectly in the future, so the else case will have to be a failure as well. This formulation is more concise. Ideally, the unwrap should have been enough, but that requires a lot of standard contract writing, which I'm putting off for now.